index.ts

```diff
@@ -53,12 +54,30 @@ export class HawkTracerClient {
             this._client = new NativeClient(option);
         }
         else {
-            this._client = new NativeClient(option.source, option.map_files);
+            this._client = new NativeClient(option.source, option.map_files, option.wait_for_server);
         }
     }
 
-    public start(): boolean {
-        return this._client.start();
+    public start(): Promise<boolean> {
+        const tryConnect = (resolve: any, reject: any) => {
+            try {
+                if (this._client.start()) {
+                    resolve(this);
+                }
+                else {
+                    setTimeout(() => {
+                        tryConnect(resolve, reject);
+                    }, 1000);
+                }
+            }
+            catch (err) {
```

Comment:
Is this necessary? I think the uncatched error automatically results in reject with the error.

Reply:
True that!